src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,128 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(last, value, null);
+        if (first == null) {
+            first = newNode;
+        } else {
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> current = findNode(index);
+        Node<T> newNode = new Node<>(current.previous, value, current);
+
+        if (current.previous == null) {
+            first = newNode;
+        } else {
+            current.previous.next = newNode;
+        }
+        current.previous = newNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
```

Comment:
The `addAll` method currently calls `add` for each element in the list, which can be inefficient. Consider optimizing this by linking nodes directly, reducing the overhead of multiple method calls.